### ISSUE TITLE
Sign in says successful, but does not actually sign you in

### ISSUE THREAD

**Issue by athomasmoz** [mozilla-mobile/firefox-tv#2862]
## Steps to reproduce

1. Open FFTV
2. Sign in to your account

### Expected behavior

You are signed into your account. You can confirm by returning to homescreen and checking toolbar

### Actual behavior

You are signed into your account. You can confirm by returning to homescreen and checking toolbar

### Device information
* Fire TV device: ?
* Latest affected Firefox version: 4.5-LAT
* Earliest affected Firefox version: ?
* Turbo Mode: ?

**Comment by mcomella** [comment]
This seems reliably reproducible if you clear data: the first time you try to sign in will fail while subsequent attempts will be successful.

We appear to extract the keys from the URL but when we get into the FxA library...
```kotlin
    fun finishAuthenticationAsync(authData: FxaAuthData): Deferred<Boolean> {
        val result = CompletableDeferred<Boolean>()

        when {
            latestAuthState == null -> {
                logger.warn("Trying to finish authentication that was never started.")
                result.complete(false)
            }
            authData.state != latestAuthState -> {
                // WE TAKE THIS BRANCH
                logger.warn("Trying to finish authentication for an invalid auth state; ignoring.")
                result.complete(false)
            }
            // ...
```

It looks like the authData we're passing in is incorrect. Theories:
- There's a bug in the FxA lib
- We're not extracting the key correctly

**Comment by mcomella** [comment]
We're parsing the URI state with the Uri lib so it's _probably_ right:
```kotlin
val uri = Uri.parse(uriStr)
            val authType = uri.getQueryParameter("action").toAuthType()
            val code = uri.getQueryParameter("code")
            val state = uri.getQueryParameter("state")
```

That leans me more towards a bug in the FxA lib.

**Comment by mcomella** [comment]
I used the debugger: on the first login, the state provided by the URL is different that the `lastAuthState` expected by the library. This looks like a bug in the FxA library.

**Comment by mcomella** [comment]
I checked out the commit before we upgraded a-c and I can’t reproduce.

**Comment by mcomella** [comment]
I checked out the commit 2 after (i.e. with the compile fixes for a-c) and I can reproduce: this is a regression from our a-c upgrade to 12.0.0. I'm debugging with Grisha on Slack.

**Comment by mcomella** [comment]
Correction: this only appears to happen from the onboarding screen but not the homescreen. It looks like the integration to the onboarding screen is causing issues. Grisha noticd `beginLogin` gets called twice, which overwrites the internal FxA library state.

**Comment by mcomella** [comment]
The issue is that we're not clearing the `compositeDisposable` the `mainActivityCommandBus` is added to.

**Comment by mcomella** [comment]
QA: please verify signing in works the first time.

**Comment by abodea** [comment]
Verified as fixed in v4.5-LAT3.
Note that I signed in more than 5 times with a fresh install and everything worked as expected.
I was able to send a tab after each first login.
